### PR TITLE
Many item calls are now isolated in a single function

### PR DESCRIPTION
**Pull request by annando**
This made some function for the term table and the thread table obsolete.

After this had been merged, the PR https://github.com/friendica/friendica-addons/pull/518 should be merged as well.

### CODE REVIEW

File: mod/tagger.php [hunk 1]
```diff
@@ -148,24 +148,16 @@ function tagger_content(App $a) {
 
 	$post_id = Item::insert($arr);
 
-	if(! $item['visible']) {
-		$r = q("UPDATE `item` SET `visible` = 1 WHERE `id` = %d AND `uid` = %d",
-			intval($item['id']),
-			intval($owner_uid)
-		);
+	if (!$item['visible']) {
+		Item::update(['visible' => true], ['id' => $item['id']]);
```
Comment (zeroadam) [inline mod/tagger.php]: Do you not need 'uid' anymore for this update?

File: mod/tagrm.php [hunk 2]
```diff
@@ -41,17 +42,12 @@ function tagrm_post(App $a) {
 
 	$tag_str = implode(',',$arr);
 
-	q("UPDATE `item` SET `tag` = '%s' WHERE `id` = %d AND `uid` = %d",
-		dbesc($tag_str),
-		intval($item),
-		intval(local_user())
-	);
+	Item::update(['tag' => $tag_str], ['id' => $item]);
```
Comment (zeroadam) [inline mod/tagrm.php]: Save 'uid' question.

File: mod/tagger.php (hunk 1 above)
Comment (annando) [reply mod/tagger.php]: The "id" field is unique. And there is no need for checking if this item really belongs to this user - this is done before.

File: mod/tagrm.php (hunk 2 above)
Comment (annando) [reply mod/tagrm.php]: Same answer ;-)

File: src/Model/Item.php [hunk 3]
```diff
@@ -156,10 +178,7 @@ public static function delete($item_id, $priority = PRIORITY_HIGH)
 
 		// If it's the parent of a comment thread, kill all the kids
 		if ($item['id'] == $item['parent']) {
-			$items = dba::select('item', ['id'], ['parent' => $item['parent']]);
-			while ($row = dba::fetch($items)) {
-				self::delete($row['id'], $priority);
-			}
+			self::delete(['parent' => $item['parent']]);
```
Comment (MrPetovan) [inline src/Model/Item.php]: Why not keep the priority in the `self::delete` call?

File: src/Model/Item.php [hunk 4]
```diff
@@ -170,6 +189,44 @@ public static function delete($item_id, $priority = PRIORITY_HIGH)
 		return true;
 	}
 
+	private static function deleteTagsFromItem($item)
```
Comment (MrPetovan) [inline src/Model/Item.php]: This function doesn't straight up delete tags, it merely updates the `tag` field from the `object` field. Please rename it to `updateTagsFromObject` or simply `updateTags`.

File: src/Model/Item.php [hunk 5]
```diff
@@ -1626,7 +1683,7 @@ public static function expire($uid, $days, $network = "", $force = false)
 				continue;
 			}
 
-			self::delete($item['id'], PRIORITY_LOW);
+			self::deleteByID($item['id'], PRIORITY_LOW);
```
Comment (MrPetovan) [inline src/Model/Item.php]: Wrong capitalization, should be `updateById`.

File: src/Model/Item.php [hunk 6]
```diff
@@ -1851,13 +1908,11 @@ public static function performLike($item_id, $verb)
...
 
-		$new_item_id = Item::insert($new_item);
+		$new_item_id = self::insert($new_item);
 
-		// @todo: Explain this block
-		if (! $item['visible']) {
-			q("UPDATE `item` SET `visible` = 1 WHERE `id` = %d",
-				intval($item['id'])
-			);
+		// If the parent item isn't visible then set it to visible
```
Comment (MrPetovan) [inline src/Model/Item.php]: We can guess this from the code, but why do we need to make it visible at this point?

File: include/text.php [hunk 7]
```diff
@@ -1897,15 +1897,8 @@ function file_tag_save_file($uid, $item, $file)
...
-		if (! stristr($r[0]['file'],'[' . file_tag_encode($file) . ']')) {
-			q("UPDATE `item` SET `file` = '%s' WHERE `id` = %d AND `uid` = %d",
-				dbesc($r[0]['file'] . '[' . file_tag_encode($file) . ']'),
-				intval($item),
-				intval($uid)
-			);
-		}
-
-		Term::insertFromFileFieldByItemId($item);
+		$fields = ['file' => $r[0]['file'] . '[' . file_tag_encode($file) . ']'];
```
Comment (MrPetovan) [inline include/text.php]: Why did you remove the duplicate check?

Comment (annando) [reply include/text.php]: The old code did some duplicate check and had only executed the command under some circumstances - but the ```Term::insertFromFileFieldByItemId``` had been executed all the time. Since that call is now no more (it is executed in the item function), I preferred to do the call every time, so that the function call to ```Term::insertFromFileFieldByItemId``` is done every time as well.

File: src/Model/Item.php (hunk 3 above)
Comment (annando) [reply src/Model/Item.php]: Because I forgot.

File: src/Model/Item.php (hunk 4 above)
Comment (annando) [reply src/Model/Item.php]: It removes (deletes) tags from items that had been added with the corresponding item array that is now about to be deleted.

File: src/Model/Item.php (hunk 6 above)
Comment (annando) [reply src/Model/Item.php]: This code had been here before. I hadn't had a deeper look at the background.

File: include/text.php (hunk 7 above)
Comment (MrPetovan) [reply include/text.php]: But this means a file can now be encoded multiple times in the `file` field, right?

File: src/Model/Item.php (hunk 4 above)
Comment (MrPetovan) [reply src/Model/Item.php]: Then I probably don't understand what tags are. I thought they were limited to a single item? From what you're saying, when we are deleting an item we a deleting tags from *other* items?

File: src/Model/Item.php (hunk 6 above)
Comment (MrPetovan) [reply src/Model/Item.php]: Fair enough.

File: include/text.php (hunk 7 above)
Comment (annando) [reply include/text.php]: Yeah ... It's maybe better to not risk this.

File: src/Model/Item.php (hunk 4 above)
Comment (annando) [reply src/Model/Item.php]: People can add tags to the items of other people.

File: src/Model/Item.php (hunk 6 above)
Comment (annando) [reply src/Model/Item.php]: Possibly this is some left over functionality from the old Diaspora module that had hidden items under certain circumstances.

File: src/Model/Item.php (hunk 4 above)
Comment (MrPetovan) [reply src/Model/Item.php]: Ah-ha! That's what I was missing.
